scripts/basic-workflow-tests.sh

```diff
@@ -231,6 +237,18 @@ test_train() {
     fi
 
     ilab model train "${TRAIN_ARGS[@]}"
+
+    if [ "$FULLTRAIN" -eq 1 ]; then
+
+        # TODO Only cuda for now
+        # the train profile specified in test_init overrides the majority of TRAIN_ARGS, including things like num_epochs. While it looks like much of those settings are being lost, they just have different values here.
+        TRAIN_ARGS=("--device=cuda" "--model-path=instructlab/granite-7b-lab" "--data-path=${SCRIPTDIR}/test-data/train_all_pruned_SDG.jsonl" "--quantize-data-type=nf4" "--4-bit-quant")
```

Comment:
just FYI -- I added another item to #1655

```
- [ ] update training-lib based training to do training based on the output of `ilab data generate` instead of a file checked in to the tree - `scripts/test-data/train_all_pruned_SDG.jsonl`
```